java/integTest/src/test/java/glide/PubSubTests.java

```diff
@@ -890,4 +879,283 @@ public void transaction_with_all_types_of_PubsubMessages(
+        assumeTrue(REDIS_VERSION.isGreaterThanOrEqualTo("7.0.0"), "This feature added in redis 7");
+
+        final GlideString channel = gs(UUID.randomUUID().toString());
+        final GlideString message = gs("1".repeat(1 << 25)); // 33MB
+
+        ArrayList<PubSubMessage> callbackMessages = new ArrayList<>();
+        final MessageCallback callback =
+                (pubSubMessage, context) -> {
+                    ArrayList<PubSubMessage> receivedMessages = (ArrayList<PubSubMessage>) context;
+                    receivedMessages.add(pubSubMessage);
+                };
+
+        Map<? extends ChannelMode, Set<GlideString>> subscriptions =
+                Map.of(PubSubClusterChannelMode.SHARDED, Set.of(channel));
+
+        var listener =
+                createClientWithSubscriptions(
+                        standalone,
+                        subscriptions,
+                        Optional.ofNullable(callback),
+                        Optional.of(callbackMessages));
+        var sender =
+                createClientWithSubscriptions(
+                        standalone,
+                        subscriptions,
+                        Optional.ofNullable(callback),
+                        Optional.of(callbackMessages));
+
+        try {
+            assertEquals(OK, ((RedisClusterClient) sender).publish(message, channel, true).get());
+
+            // Allow the message to propagate.
+            Thread.sleep(MESSAGE_DELIVERY_DELAY);
+
+            assertEquals(1, callbackMessages.size());
+            assertEquals(message, callbackMessages.get(0).getMessage());
+        } finally {
+            if (!standalone) {
+                // Since all tests run on the same cluster, when closing the client, garbage collector can
+                // be called
+                // after another test will start running.
+                // In cluster mode, we check how many subscriptions received the message.
+                // So to avoid flakiness, we make sure to unsubscribe from the channels.
+                ((RedisClusterClient) listener)
+                        .customCommand(new String[] {"UNSUBSCRIBE", channel.toString()});
+            }
+        }
+    }
+
+    /** Test the behavior if the callback supplied to a subscription throws an uncaught exception. */
+    @SneakyThrows
+    @ParameterizedTest(name = "standalone = {0}")
+    @ValueSource(booleans = {true, false})
+    @Disabled(
+            "This test is for demonstrating behavior when there's an exception from a user callback and"
+                    + " will always fail.")
+    public void pubsub_test_callback_exception(boolean standalone) {
+        final GlideString channel = gs(UUID.randomUUID().toString());
+        final GlideString message = gs("message");
+
+        ArrayList<PubSubMessage> callbackMessages = new ArrayList<>();
+        final MessageCallback callback =
+                (pubSubMessage, context) -> {
```

Comment:
If you update it to throw on every second/fifth message only, you can make test pass. It will also prove that client keeps working after having an error.
```
if (msg.contains("throwme"))
    throw new err(msg);
```
You can also execute any simple command after that to confirm that client is operational:
```
client.set(key, value)
value == client.get(key)
```

Reply:
I had it publish another message after an exception-throwing one.